Gems/LmbrCentral/Code/Source/Shape/EditorSplineComponentMode.cpp

```diff
@@ -35,6 +36,40 @@ namespace LmbrCentral
         m_vertexSelection.Destroy();
     }
 
+    void EditorSplineComponentMode::Reflect(AZ::ReflectContext* context)
+    {
+        if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serializeContext->Class<EditorSplineComponentMode, EditorBaseComponentMode>(AZ::Internal::NullFactory::GetInstance());
+        }
+    }
+
+    void EditorSplineComponentMode::RegisterActions()
+    {
+        AzToolsFramework::EditorVertexSelection::RegisterEditorVertexSelectionActions();
+    }
+
+    void EditorSplineComponentMode::BindActionsToModes()
+    {
+        auto actionManagerInterface = AZ::Interface<AzToolsFramework::ActionManagerInterface>::Get();
+        AZ_Assert(actionManagerInterface, "EditorVertexSelection - could not get ActionManagerInterface on RegisterActions.");
+
+        AZ::SerializeContext* serializeContext = nullptr;
+        AZ::ComponentApplicationBus::BroadcastResult(serializeContext, &AZ::ComponentApplicationRequests::GetSerializeContext);
+        
+        AZStd::string modeIdentifier = AZStd::string::format(
+            "o3de.context.mode.%s", serializeContext->FindClassData(azrtti_typeid<EditorSplineComponentMode>())->m_name);
+
+        actionManagerInterface->AssignModeToAction(modeIdentifier, "o3de.action.vertexSelection.duplicate");
```

Comment:
suggestion: The identifiers should likely be moved to constants as they crop up in a few places